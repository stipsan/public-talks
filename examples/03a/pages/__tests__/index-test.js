import { mount, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import App from '../index'

describe('With Enzyme', () => {
  describe('enzyme-to-json in shallow mode', () => {
    it('App renders correctly', () => {
      const app = shallow(<App />)

      expect(app).toMatchSnapshot()
    })
  })
  describe('enzyme-to-json in deep mode', () => {
    it('App renders correctly', () => {
      const app = mount(<App />)

      expect(toJson(app, { mode: 'deep' })).toMatchSnapshot()
    })
  })
})
