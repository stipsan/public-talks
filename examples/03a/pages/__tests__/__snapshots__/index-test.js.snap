// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Enzyme enzyme-to-json in deep mode App renders correctly 1`] = `
<div>
  <p>
    Hello 
    ReactJS Oslo!
    !
  </p>
  <footer>
    Â© 
    2018
     
    ACME Inc.
  </footer>
</div>
`;

exports[`With Enzyme enzyme-to-json in shallow mode App renders correctly 1`] = `
<div>
  <Hello
    entity="ReactJS Oslo!"
  />
  <Footer
    whoGotAllTheRights="ACME Inc."
    year={2018}
  />
</div>
`;
